src/librustc_trans/mir/mod.rs

```diff
@@ -384,9 +385,13 @@ fn arg_local_refs<'a, 'tcx>(bcx: &Builder<'a, 'tcx>,
                 _ => bug!("spread argument isn't a tuple?!")
             };
 
+            let variant = match *bcx.ccx.layout_of(arg_ty).layout {
+                layout::Univariant { ref variant, .. } => variant,
+                _ => bug!("spread argument is only supposed to be Univariant")
+            };
             let lvalue = LvalueRef::alloca(bcx, arg_ty, &format!("arg{}", arg_index));
             for (i, &tupled_arg_ty) in tupled_arg_tys.iter().enumerate() {
-                let dst = bcx.struct_gep(lvalue.llval, i);
+                let dst = bcx.struct_gep(lvalue.llval, adt::struct_llfields_index(variant, i));
```

Comment:
I would have to see the rebased version to tell you if what you did here is correct.